### PR TITLE
2080 - fix sonarqube security hotspots

### PR DESCRIPTION
**Pull request by cogross**
Fixed some things that SonarQube found that were potential security issues.
Includes:
 - SQL injections
 - regex polynomial runtimes
 - pseudorandom # generators
 - cookie configurations

### CODE REVIEW

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/summary/util/MetricRuleUtil.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public final class MetricRuleUtil {
     private MetricRuleUtil() {}
 
     private static final Matcher validRule = Pattern.compile("[a-z_0-9]{3,30}").matcher("");;
-    private static final Matcher longNumericSequence = Pattern.compile(".*[\\d]{5,}.*").matcher("");;
+    private static final Matcher longNumericSequence = Pattern.compile("[^\\d]*[\\d]{5,}[^\\d]*").matcher("");;
```
Comment (cogross) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/summary/util/MetricRuleUtil.java]: Fixed regex to reduce chance of polynomial runtime.

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java [hunk 2]
```diff
@@ -689,7 +689,7 @@ protected Configuration parseArguments(String[] args, Configuration conf) throws
             } else if (args[i].equals("-cacheBaseDir")) {
                 cacheBaseDir = args[++i];
             } else if (args[i].equals("-cacheJars")) {
-                String[] jars = StringUtils.trimAndRemoveEmptyStrings(args[++i].split("\\s*,\\s*"));
+                String[] jars = StringUtils.trimAndRemoveEmptyStrings(args[++i].replaceAll("\\s*", " ").split("\\s?,\\s?"));
```
Comment (cogross) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java]: Fixed regex to reduce chance of polynomial runtime.

File: warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java [hunk 3]
```diff
@@ -78,7 +79,7 @@ public class VisitorFunction implements Function<ScannerChunk,ScannerChunk> {
     protected Set<String> indexedFields;
     protected Set<String> indexOnlyFields;
     protected Set<String> nonEventFields;
-    protected Random random = new Random();
+    protected Random random = new SecureRandom();
```
Comment (cogross) [inline warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java:81]: Safer random number generator.

File: warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java [hunk 4]
```diff
@@ -69,7 +70,7 @@ public abstract class DatawaveFieldIndexCachingIteratorJexl extends WrappingIter
     public static final Text ANY_FINAME = new Text("fi\0" + Constants.ANY_FIELD);
     public static final Text FI_START = new Text("fi\0");
     public static final Text FI_END = new Text("fi\0~");
-    public static final Random RANDOM = new Random();
+    public static final Random RANDOM = new SecureRandom();
```
Comment (cogross) [inline warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java:73]: Safer random number generator.

File: web-services/cached-results/src/main/java/datawave/webservice/query/database/CachedResultsCleanupBean.java [hunk 5]
```diff
@@ -90,15 +90,15 @@ public void cleanup() {
                 while (rs.next()) {
                     String objectName = CachedResultsParameters.validate(rs.getString(1));
                     // Drop the table
-                    String dropTable = "DROP TABLE " + objectName;
+                    String dropTable = String.format("DROP TABLE %s", objectName);
```
Comment (cogross) [inline web-services/cached-results/src/main/java/datawave/webservice/query/database/CachedResultsCleanupBean.java:93]: Cleaner dynamic SQL generation.

File: web-services/cached-results/src/main/java/datawave/webservice/results/cached/CachedResultsBean.java [hunk 6]
```diff
@@ -1140,13 +1140,13 @@ public CachedResultsDescribeResponse describe(@PathParam("id") @Required("id") S
             List<String> columns = new ArrayList<>();
             Integer numRows = null;
             try (Connection con = ds.getConnection(); Statement s = con.createStatement()) {
-                try (ResultSet rs = s.executeQuery("select count(*) from " + view)) {
+                try (ResultSet rs = s.executeQuery(String.format("select count(*) from %s", view))) {
```
Comment (cogross) [inline web-services/cached-results/src/main/java/datawave/webservice/results/cached/CachedResultsBean.java:1140]: Cleaner dynamic SQL generation.

File: web-services/cached-results/src/main/java/datawave/webservice/results/cached/CachedRunningQuery.java [hunk 7]
```diff
@@ -555,7 +555,7 @@ public boolean isActivated() {
     private List<String> getViewColumnNames(Connection connection, String view) throws SQLException {
         CachedResultsParameters.validate(view);
         List<String> columns = new ArrayList<>();
-        try (Statement s = connection.createStatement(); ResultSet rs = s.executeQuery("show columns from " + view)) {
+        try (Statement s = connection.createStatement(); ResultSet rs = s.executeQuery(String.format("show columns from %s", view))) {
```
Comment (cogross) [inline web-services/cached-results/src/main/java/datawave/webservice/results/cached/CachedRunningQuery.java:558]: Cleaner dynamic SQL generation.

File: web-services/common-util/src/main/java/datawave/resteasy/interceptor/CreateQuerySessionIDFilter.java [hunk 8]
```diff
@@ -58,7 +58,7 @@ public void filter(ContainerRequestContext request, ContainerResponseContext res
 
         if (setCookie) {
             response.getHeaders().add(HttpHeaderNames.SET_COOKIE,
-                            new NewCookie(Constants.QUERY_COOKIE_NAME, cookieValue, path + id, null, null, NewCookie.DEFAULT_MAX_AGE, false));
+                            new NewCookie(Constants.QUERY_COOKIE_NAME, cookieValue, path + id, null, null, NewCookie.DEFAULT_MAX_AGE, true, true));
```
Comment (cogross) [inline web-services/common-util/src/main/java/datawave/resteasy/interceptor/CreateQuerySessionIDFilter.java]: Cookies without HttpOnly=true, secure=true aren't safe.
While the tests pass, I'm not sure how to test this directly.  Perhaps the integration tests will help.

File: web-services/security/src/main/java/datawave/security/authorization/simple/DatabaseUserService.java [hunk 9]
```diff
@@ -120,7 +120,7 @@ public DatabaseUserService(@ConfigProperty(name = "dw.databaseUsersService.users
     public void setup() {
         try (Connection c = ds.getConnection();
                         Statement s = c.createStatement();
-                        ResultSet rs = s.executeQuery("SELECT role, auth FROM " + mappingTableName)) {
+                        ResultSet rs = s.executeQuery(String.format("SELECT role, auth FROM %s", mappingTableName))) {
```
Comment (cogross) [inline web-services/security/src/main/java/datawave/security/authorization/simple/DatabaseUserService.java:123]: Cleaner dynamic SQL generation.

**Review by awrigh11 — DISMISSED** [review]

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/summary/util/MetricRuleUtil.java (hunk 1 above)
Comment (cogross) [reply warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/summary/util/MetricRuleUtil.java]: Rolled back.

File: warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java [hunk 10]
```diff
@@ -682,7 +682,7 @@ protected Configuration parseArguments(String[] args, Configuration conf) throws
             } else if (args[i].equals("-cacheBaseDir")) {
                 cacheBaseDir = args[++i];
             } else if (args[i].equals("-cacheJars")) {
-                String[] jars = StringUtils.trimAndRemoveEmptyStrings(args[++i].split("\\s*,\\s*"));
+                String[] jars = StringUtils.trimAndRemoveEmptyStrings(args[++i].replaceAll("\\s+", "").split("\\s{0,1},\\s{0,1}"));
```
Comment (matthpeterson) [inline warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java]: Doesn't `.replaceAll("\\s+", "")` make `split(",")` sufficient?

Comment (cogross) [reply warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/IngestJob.java]: it does -- fixed.

**Review by ivakegg — APPROVED** [review]

**Review by matthpeterson — APPROVED** [review]

**Review by keith-ratcliffe — APPROVED** [review]

**Pull request merged**